openlibrary/plugins/importapi/tests/test_code.py

```diff
@@ -53,6 +55,105 @@ def test_get_ia_record(monkeypatch, mock_site, add_languages) -> None:  # noqa F
+    """Some IA records have a single string for the publisher, but it should be a list."""
+    monkeypatch.setattr(web, "ctx", web.storage())
+    web.ctx.lang = "eng"
+    web.ctx.site = mock_site
+
+    # Testing publisher is a string.
+    ia_metadata = {
+        "creator": "The Author",
+        "date": "2013",
+        "identifier": "ia_frisian001",
+        "publisher": "The Publisher",
+        "title": "Frisian is Fun",
+    }
+
+    expected_result = {
+        "authors": [{"name": "The Author"}],
+        "publish_date": "2013",
+        "publishers": ["The Publisher"],
+        "title": "Frisian is Fun",
+    }
+
+    result = code.ia_importapi.get_ia_record(ia_metadata)
+    assert result == expected_result
+
+    # Testing publisher is a list.
+    ia_metadata = {
+        "creator": "The Author",
+        "date": "2013",
+        "identifier": "ia_frisian001",
+        "publisher": "The Publisher",
+        "title": "Frisian is Fun",
+    }
+
+    expected_result = {
+        "authors": [{"name": "The Author"}],
+        "publish_date": "2013",
+        "publishers": ["The Publisher"],
+        "title": "Frisian is Fun",
+    }
+
+    result = code.ia_importapi.get_ia_record(ia_metadata)
+    assert result == expected_result
+
+
+def test_get_ia_record_handles_isbn_10_and_isbn_13() -> None:
+    """
+    IA records have an `isbn` field with potentially both ISBN 10 and ISBN 13.
+    But Open Library needs them sorted into `isbn_10` and `isbn_13` lists.
+    """
+    ia_metadata = {
+        "creator": "The Author",
+        "date": "2013",
+        "identifier": "ia_frisian001",
+        "isbn": ["1280711191", "9781280711190", "1576079457", "9781576079454"],
+        "publisher": "The Publisher",
+        "title": "Frisian is Fun",
+    }
+
+    expected_result = {
+        "authors": [{"name": "The Author"}],
+        "publish_date": "2013",
+        "isbn_10": ["1280711191", "1576079457"],
+        "isbn_13": ["9781280711190", "9781576079454"],
+        "publishers": ["The Publisher"],
+        "title": "Frisian is Fun",
+    }
+
+    result = code.ia_importapi.get_ia_record(ia_metadata)
+    assert result == expected_result
+
+
+def test_get_ia_record_handles_publishers_with_places() -> None:
+    """
+    Some IA records have the format:
+        "publisher": "New York : Simon & Schuster"
+
+    Split on ` : ` and put the place into `publish_places: list[str]`
+    """
+    ia_metadata = {
+        "creator": "The Author",
+        "date": "2013",
+        "identifier": "ia_frisian001",
+        "publisher": "New York : Simon & Schuster",
+        "title": "Frisian is Fun",
+    }
+
+    expected_result = {
+        "authors": [{"name": "The Author"}],
+        "publish_date": "2013",
+        "publishers": ["Simon & Schuster"],
+        "publish_places": ["New York"],
+        "title": "Frisian is Fun",
+    }
+
+    result = code.ia_importapi.get_ia_record(ia_metadata)
+    assert result == expected_result
+
+
```

Comment:
Change these to only test the two inner functions -- `get_location_and_publisher` , `get_isbn_10_and_13`

Reply:
Seems I already had separate tests for these functions. Go figure!